cypress/e2e/APIRequests.cy.ts

```diff
@@ -46,38 +61,103 @@ describe('API request', () => {
       emptyAssessmentToolOptions
     ).as('getAssessmentToolOptions');
     cy.visit('/');
-    cy.wait('@getDiagnosisOptions');
-    cy.get('.notistack-SnackbarContainer').should('contain', 'No Diagnosis options were available');
+    cy.wait('@getAssessmentToolOptions');
+
     cy.get('.notistack-SnackbarContainer').should(
       'contain',
       'No Assessment tool options were available'
     );
   });
-  it('Failed responses for diagnosis and assessment make an error toast appear', () => {
+});
+
+describe('Partially successful API attribute responses', () => {
+  beforeEach(() => {
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/nodes/',
+      },
+      nodeOptions
+    ).as('getNodes');
     cy.intercept(
       {
         method: 'GET',
         url: '/attributes/nb:Diagnosis',
       },
-      { statusCode: 500 }
+      partiallyFailedDiagnosisToolOptions
     ).as('getDiagnosisOptions');
     cy.intercept(
       {
         method: 'GET',
         url: '/attributes/nb:Assessment',
       },
-      { statusCode: 500 }
+      partiallyFailedAssessmentToolOptions
     ).as('getAssessmentToolOptions');
     cy.visit('/');
-    cy.wait('@getDiagnosisOptions');
-    cy.get('.notistack-SnackbarContainer').should(
-      'contain',
-      'Failed to retrieve Diagnosis options'
-    );
-    cy.wait('@getAssessmentToolOptions');
-    cy.get('.notistack-SnackbarContainer').should(
-      'contain',
-      'Failed to retrieve Assessment tool options'
-    );
+    cy.wait(['@getNodes', '@getDiagnosisOptions', '@getAssessmentToolOptions']);
+  });
+  it('Shows warning for node that failed Assessment tool option request', () => {
+    cy.get('.notistack-SnackbarContainer').should('contain', 'NoAssessmentNode');
+  });
+  it('Shows warning for node that failed Diagnosis option request', () => {
+    cy.get('.notistack-SnackbarContainer').should('contain', 'NoDiagnosisNode');
+  });
+});
+
+describe('Failed API attribute responses', () => {
+  beforeEach(() => {
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/nodes/',
+      },
+      nodeOptions
+    ).as('getNodes');
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/attributes/nb:Diagnosis',
+      },
+      failedDiagnosisToolOptions
+    ).as('getDiagnosisOptions');
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/attributes/nb:Assessment',
+      },
+      failedAssessmentToolOptions
+    ).as('getAssessmentToolOptions');
+    cy.visit('/');
+    cy.wait(['@getNodes', '@getDiagnosisOptions', '@getAssessmentToolOptions']);
+  });
+  it('Shows error toast for failed Assessment tool options', () => {});
+  it('Shows error toast for failed Diagnosis options', () => {});
```

Comment:
May have forgotten to address the `failure` case.

Reply:
Just to be clear: you mean I forgot to add a test for this case, correct? Not that the handling of such a case is missing from the app itself?

Reply:
I believe it's both.

Reply:
From the issue description

> complete failure
   none of the nodes have responded
    -> do not show any results, instead show an error message in the results area
    -> also show an error toast and say that all nodes have failed.

Reply:
OK, thanks for the hint @rmanaem. Turns out this was a bigger change - so probably best to review again!